tests/integration/test_kubectl_delete.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from helpers import get_config_values
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+app_name = METADATA["name"]
+config = {"scrape_interval": "15s", "scrape_timeout": "10s"}
+
+
+@pytest.mark.abort_on_fail
+async def test_deploy_from_local_path(ops_test, charm_under_test):
+    """Deploy the charm-under-test."""
+    logger.debug("deploy local charm")
+
+    resources = {"unused-image": METADATA["resources"]["unused-image"]["upstream-source"]}
+    await ops_test.model.deploy(charm_under_test, application_name=app_name, resources=resources)
+
+    await ops_test.model.applications[app_name].set_config(config)
+    await ops_test.model.wait_for_idle(apps=[app_name], status="blocked", timeout=1000)
+
+
+@pytest.mark.abort_on_fail
+async def test_kubectl_delete_pod(ops_test):
+    pod_name = f"{app_name}-0"
+
+    cmd = [
+        "sg",
+        "microk8s",
+        "-c",
+        " ".join(["microk8s.kubectl", "delete", "pod", "-n", ops_test.model_name, pod_name]),
+    ]
+
+    logger.debug(
+        "Removing pod '%s' from model '%s' with cmd: %s", pod_name, ops_test.model_name, cmd
+    )
+
+    retcode, stdout, stderr = await ops_test.run(*cmd)
+    assert retcode == 0, f"kubectl failed: {(stderr or stdout).strip()}"
+    logger.debug(stdout)
+    await ops_test.model.block_until(lambda: len(ops_test.model.applications[app_name].units) > 0)
+    await ops_test.model.wait_for_idle(apps=[app_name], status="blocked", timeout=1000)
+
+    assert (await get_config_values(ops_test, app_name)).items() >= config.items()
```

Comment:
It is not obvious why a "kubectl_delete_pod" test is checking for config values ?

Reply:
That is the heart of the test and has already proven useful: config values that were set before, should be present after.
This test detected the underscore issue right away.

Reply:
Please see my comments elsewhere in this PR regarding `is_healthy(). What I am trying to say is this -

If your test was named `test_units_are_restarted_on_pod_delete` or `test_config_values_are_retained_after_pod_restart` it would have been obvious what is being tested for. Further those two tests would mean two different things though both are possibly not necessary. 

Since your test was named `test_kubectl_delete_pod` and you were checking config values it was not obvious to me what was being tested for.

Does this sound like a reasonable doubt ?

The benefit of making it very obvious what is being tested for is so that when the test fails the developer immediately knows what is failing.

Reply:
Good point! Renamed to `test_config_values_are_retained_after_pod_deleted_and_restarted`